src/uct/cuda/cuda_ipc/cuda_ipc_ep.h

```diff
@@ -9,14 +9,44 @@
 #include <uct/api/uct.h>
 #include <uct/base/uct_iface.h>
 #include <ucs/type/class.h>
+#include <ucs/datastruct/khash.h>
+#include "cuda_ipc_md.h"
 
+typedef struct uct_cuda_ipc_rem_seg uct_cuda_ipc_rem_seg_t;
+
+struct uct_cuda_ipc_rem_seg {
+    uct_cuda_ipc_rem_seg_t *next;
+    CUipcMemHandle         ph;         /* Memory handle of GPU memory */
+    CUdeviceptr            d_bptr;     /* Allocation base address */
+    size_t                 b_len;      /* Allocation size */
+    int                    dev_num;    /* GPU Device number */
+};
+
+static inline khint_t uct_cuda_ipc_memh_hash_func(CUipcMemHandle seg)
+{
+    int hash_val = 7;
+    int i;
+
+    for (i = 0; i < sizeof(seg); i++) {
+        hash_val = hash_val*31 + seg.reserved[i];
+    }
+
+    return (khint_t) (hash_val);
```

Comment:
casting probably notrneeded here

Reply:
@yosefe I've fixed this. Does the hash function logic look correct to you? Is this needed or is there something simpler that works? From the code, you can see that I use the key just for comparison but unfortunately, the key is a string of a fixed size else if it was an integer, I'd simply use an identity function. Lmk what you think.